rust_src/src/buffers.rs

```diff
@@ -488,6 +486,78 @@ fn assoc_ignore_text_properties(key: LispObject, list: LispObject) -> LispObject
     }
 }
 
+pub enum LispBufferOrName {
+    Buffer(LispBufferRef),
+}
+
+impl LispBufferOrName {
+    pub fn unwrap(self) -> LispBufferRef {
+        match self {
+            LispBufferOrName::Buffer(b) => b,
+        }
+    }
+}
+
+fn buffer_or_by_name(buffer_or_name: LispObject) -> LispBufferRef {
+    if buffer_or_name.is_nil() {
+        nsberror(buffer_or_name);
+    }
+
+    if let Some(b) = buffer_or_name.as_buffer() {
+        b
+    } else {
+        buffer_or_name.as_string_or_error();
+        let b = cdr(assoc_ignore_text_properties(buffer_or_name, unsafe {
+            Vbuffer_alist
+        }));
+
+        if b.is_nil() {
+            nsberror(buffer_or_name);
+        }
+
+        b.as_buffer_or_error()
+    }
+}
+
+impl From<LispObject> for LispBufferOrName {
+    fn from(buffer_or_name: LispObject) -> Self {
+        let buffer = buffer_or_by_name(buffer_or_name);
+        LispBufferOrName::Buffer(buffer)
+    }
+}
+
+pub enum LispBufferOrCurrent {
+    Buffer(LispBufferRef),
+}
+
+impl LispBufferOrCurrent {
+    pub fn unwrap(self) -> LispBufferRef {
+        match self {
+            LispBufferOrCurrent::Buffer(b) => b,
+        }
+    }
+}
+
+impl From<LispObject> for LispBufferOrCurrent {
+    fn from(buffer_or_name: LispObject) -> Self {
+        let obj = if buffer_or_name.is_nil() {
+            current_buffer()
+        } else {
+            buffer_or_name
+        };
+        let buffer = buffer_or_by_name(obj);
+        LispBufferOrCurrent::Buffer(buffer)
+    }
+}
+
+pub fn lispbuffer_or_by_name(buffer_or_name: LispObject) -> LispObject {
```

Comment:
We should choose a name that suggests that it's similiar to get_buffer except it returns nil instead of throwing an error for nil. Maybe get_buffer and get_buffer_or_error. Would be consistent.

Reply:
Yeah. Or I might just drop the function for now.